ethers-core/src/types/transaction/optimism.rs

```diff
@@ -48,14 +48,14 @@ impl DepositTransaction {
         let mut rlp = RlpStream::new();
         rlp.begin_list(NUM_TX_FIELDS);
 
-        rlp_opt(&mut rlp, &self.source_hash);
-        rlp.append(&self.tx.from);
+        rlp.append(&self.source_hash.unwrap_or_default());
+        rlp.append(&self.tx.from.unwrap_or_default());
         rlp_opt(&mut rlp, &self.tx.to);
         rlp_opt(&mut rlp, &self.mint);
-        rlp.append(&self.tx.value);
-        rlp.append(&self.tx.gas);
-        rlp_opt(&mut rlp, &self.is_system_tx);
-        rlp_opt(&mut rlp, &self.tx.data.as_deref());
+        rlp.append(&self.tx.value.unwrap_or_default());
+        rlp.append(&self.tx.gas.unwrap_or_default().as_u64());
```

Comment:
same here, rlp_opt would fallback to an empty string and `0` is also encoded as empty string code I believe.

but this is better.